cli/bookkeeper/bookkeeper.go

```diff
@@ -15,19 +18,51 @@ import (
 	"github.com/urfave/cli"
 )
 
-func makeBookkeeperTransaction(pubkey *crypto.PubKey, op bool, cert []byte) (string, error) {
-	tx, _ := transaction.NewBookKeeperTransaction(pubkey, op, cert)
+func makeBookkeeperTransaction(pubkey *crypto.PubKey, op bool, cert []byte, issuer *account.Account) (string, error) {
+	tx, _ := transaction.NewBookKeeperTransaction(pubkey, op, cert, issuer.PubKey())
 	attr := transaction.NewTxAttribute(transaction.Nonce, []byte(strconv.FormatInt(rand.Int63(), 10)))
 	tx.Attributes = make([]*transaction.TxAttribute, 0)
 	tx.Attributes = append(tx.Attributes, &attr)
+	if err := signTransaction(issuer, tx); err != nil {
+		fmt.Println("Sign regist transaction failed.")
+		return "", err
+	}
 	var buffer bytes.Buffer
 	if err := tx.Serialize(&buffer); err != nil {
-		fmt.Println("serialize bookkeeper transaction failed")
+		fmt.Println("Serialize bookkeeper transaction failed.")
 		return "", err
 	}
 	return hex.EncodeToString(buffer.Bytes()), nil
 }
 
+func newContractContextWithoutProgramHashes(data signature.SignableData) *contract.ContractContext {
+	return &contract.ContractContext{
+		Data:       data,
+		Codes:      make([][]byte, 1),
+		Parameters: make([][][]byte, 1),
+	}
+}
+
+func signTransaction(signer *account.Account, tx *transaction.Transaction) error {
+	signature, err := signature.SignBySigner(tx, signer)
+	if err != nil {
+		fmt.Println("SignBySigner failed.")
+		return err
+	}
+	transactionContract, err := contract.CreateSignatureContract(signer.PubKey())
+	if err != nil {
+		fmt.Println("CreateSignatureContract failed.")
+		return err
+	}
+	transactionContractContext := newContractContextWithoutProgramHashes(tx)
```

Comment:
Could use contract.NewContractContext instead, since this transaction doesn't require local store